### PR TITLE
CD-173: XClarity (Lenovo) Cooling System Health

### PR DESCRIPTION
**Pull request by mariemm97**

### CODE REVIEW

File: examples/ssh/xclarity_cooling_system_health.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    },
+    "tags": [
+        "xclarity",
+        "lenovo",
+        "cooling",
+        "health",
+        "speed",
+        "status"
+    ],
+    "sample_period": 1800,
```
Comment (kgskgs) [inline examples/ssh/xclarity_cooling_system_health.json:23]: change to sample_period_s

File: examples/ssh/xclarity_cooling_system_health.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "version": "1.0.0",
+    "name": "XClarity Cooling System Health",
+    "description": "Monitor cooling system health on Lenovo XClarity server",
+    "category": "Software Integration",
+    "requirements": {
+        "credentials": true,
+        "sandbox_version": "1.12",
+        "others": [
+            "SSH Enabled",
+            "The keyboard_interactive option can be true or false depending on the XClarity version"
```
Comment (kgskgs) [inline examples/ssh/xclarity_cooling_system_health.json:12]: provides no information

File: examples/ssh/xclarity_cooling_system_health.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    "timeout": 100000,
+    "keyboard_interactive": true
+};
+
+// Create a custom table to store cooling system health information
+var coolingSystemTable = D.createTable(
+    "Cooling System Health",
+    [
+        { label: "Speed", unit: "RPM", valueType: D.valueType.NUMBER },
+        { label: "Speed (% of maximum)", unit: "%", valueType: D.valueType.NUMBER },
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_cooling_system_health.js:31]: Just name then "Speed Value" and "Speed Percentage"

File: examples/ssh/xclarity_cooling_system_health.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{
+    "logo": "lenovo-xclarity.svg",
+    "version": "1.0.0",
+    "name": "XClarity Cooling System Health",
+    "description": "Monitor cooling system health on Lenovo XClarity server",
+    "category": "Software Integration",
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_cooling_system_health.json:6]: hardware

File: examples/ssh/xclarity_cooling_system_health.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+// Parse and insert cooling system health data into the custom table
+function parseData(output) {
+    var lines = output.split("\n");
+    var recordIdReservedWords = ['\\?', '\\*', '\\%', 'table', 'column', 'history'];
+    var recordIdSanitisationRegex = new RegExp(recordIdReservedWords.join('|'), 'g');
+    for (var i = 2; i < lines.length-1; i++) {
+        var line = lines[i].trim();
+        var values = line.split(/\s+/);
+        var recordId = (values[0] + values[1] + "-" + values[2]).replace(recordIdSanitisationRegex, '').slice(0, 50);
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_cooling_system_health.js:96]: also lower case it

File: examples/ssh/xclarity_cooling_system_health.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+ */
+function validate() {
+    executeCommand(command)
+        .then(parseValidateOutput)
+        .then(D.success)
+        .catch(checkSshError);
+}
+
+function parseValidateOutput(output) {
+    if (output.trim() !== "") {
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_cooling_system_health.js:72]: validate properly if the command is supported

**Pull request merged**